### PR TITLE
chore: enable CSE ACR connectivity check only for AKS

### LINKED ISSUE #2212: Failure when processing extension 'vmssCSE'. Exit status=50
**Describe the bug**

We have problems to deploy with aks-engine because some agent pools can't access to internet during bootstrapping. This is the error:

`
Error: Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details." Details=[{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"VMExtensionProvisioningError\",\r\n        \"message\": \"VM has reported a failure when processing extension 'vmssCSE'. Error message: \\\"Enable failed: failed to execute command: command terminated with exit status=50\\n[stdout]\\nThu Oct 24 14:18:01 UTC 2019,k8s-genm-21270049-vmss000000\\n\\n[stderr]\\n\\\".\"\r\n      }\r\n    ]\r\n  }\r\n}"}]
`

Checking documentation and code, I have seen that node executes some `nc` command to check connectivity with some docker registries. If these commands fail, error 50 is returned.  Also, I have connected manually to a failed node and it had connectivity issues. 

I don't know if it's an aks-engine issue or an Azure cloud issue. Are you having similar issues?

**Steps To Reproduce**
Execute `aks-engine deploy` command. 

**Expected behavior**
Correct deployment. 

**AKS Engine version**
0.42.2

**Kubernetes version**
1.14.7

**Additional context**

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

AKS cluster bootstrapping depends upon ACR, so we need to ensure it includes that connectivity check before running CSE.

**Issue Fixed**:

Fixes #2212 

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/virtualmachinescalesets.go [hunk 1]
```diff
@@ -298,13 +298,13 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 		ncBinary = "ncat"
 	}
 	// TODO The AzureStack constraint has to be relaxed, it should only apply to *disconnected* instances
-	if !cs.Properties.FeatureFlags.IsFeatureEnabled("BlockOutboundInternet") && !cs.Properties.IsAzureStackCloud() {
+	if !cs.Properties.FeatureFlags.IsFeatureEnabled("BlockOutboundInternet") && !cs.Properties.IsAzureStackCloud() && cs.Properties.IsHostedMasterProfile() {
```
Comment (CecileRobertMichon) [inline pkg/engine/virtualmachinescalesets.go]: I think we can remove the `!cs.Properties.IsAzureStackCloud()` condition here

File: pkg/engine/vmextensions.go [hunk 2]
```diff
@@ -68,13 +68,13 @@ func CreateCustomScriptExtension(cs *api.ContainerService) VirtualMachineExtensi
 	}
 
 	// TODO The AzureStack constraint has to be relaxed, it should only apply to *disconnected* instances
-	if !cs.Properties.FeatureFlags.IsFeatureEnabled("BlockOutboundInternet") && !cs.Properties.IsAzureStackCloud() {
+	if !cs.Properties.FeatureFlags.IsFeatureEnabled("BlockOutboundInternet") && !cs.Properties.IsAzureStackCloud() && cs.Properties.IsHostedMasterProfile() {
```
Comment (CecileRobertMichon) [inline pkg/engine/vmextensions.go]: ditto

**Comment by arodriguezdlc** [comment]
⚠️ I am deploying k8s with this fix. The error 50 doesn't appear (as expected), but still there are connectivity issues trying to download docker images. For example:

```
Warning  Failed     54m (x5 over 59m)      kubelet, k8s-common-21270049-vmss000000  Failed to pull image "calico/pod2daemon-flexvol:v3.8.0": rpc error: code = Unknown desc = Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Also I have checked a deployment using aks-engine v0.40.0 and we don't have this error. It started after upgrading to v0.41.4. Maybe there is a bug introduced in a commit between both versions that is producing this network issues.

**Comment by CecileRobertMichon** [comment]
@arodriguezdlc are you still running into the error? We aren't seeing any regression signals in tests but it's possible that there was a temporary network issue or docker outage... Which region was this in?

**Comment by arodriguezdlc** [comment]
Currently I don't having this error. Maybe Azure cloud had problems last week. 

However, last week aks-engine 0.40.0 didn't have errors, but 0.42.0 yes. So maybe there's a new feature that depends on the affected services of Azure.

**Pull request merged**